neutrons/SNAPRed
# toss an exception stating which group contains a nan metric

## Description of work

This will toss a stopping exception in the Calibration workflow if it detects nans in the metrics.

This is typically indicative of a fit failing, and thus the user should retweak params till fits succeed.

## Explanation of work

It just scans for nans in the end result of metric and catalogs in an error which groups are flagrant

## To test

Probably easiest to do out of your own Calibration folder.
Even easier if you wait till FWHM gets exposed on the front end.

(if having trouble reproducing use the calibration from the mainline Calibration_dynamic) 

(make sure in your calibration's CalibrationParameters.json that FWHM is 2 on both multipliers)
Try to calibrate with:
49525
Convergence: 0.5
Peak Intensity: 0
Bin: 7
Diamond
Column
Lorentzian

Observe it failing

start again with your calibration's CalibrationParameters.json FWHM set to 3 each

Observe it passing.

## Link to EWM item


[EWM#<ticket_number>](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=<ticket_number>)
